src/tr2/game/render/hwr.c

```diff
@@ -599,35 +589,31 @@ static const int16_t *M_InsertObjectG3_Sorted(
             continue;
         }
 
-        const RGB_888 *const color = &g_GamePalette16[color_idx >> 8];
+        const RGB_888 *const color = &g_GamePalette16[face->texture >> 8];
```

Comment:
Maybe we could make it an union and still refer to it as `face->color_idx`?

Reply:
Good idea, shall I push here or do separately and update TR1 as well?

Reply:
It's OK to push here and do an update to TR1 as well while at it.

Reply:
Ok, will do so tomorrow morning 👌

Reply:
@rr- It turns out this would be overly complicated with the current TR1 injection module, which does things like "get the texture on face N of mesh K of model M and use it on face O of mesh J of model P. We'd have to introduce separate ways of saying if it's flat, use the color_idx, else texture_idx, and so it's a lot of extra code.
Maybe during the injection format refactor we can come up with something better here.

For now, I've restored creating explicit `color_idx` variables for clarity in the flat face rendering functions.